### ISSUE TITLE
Add REST API

### ISSUE THREAD

**Issue by mrmn2** [mrmn2/PdfDing#7]

**Comment by wh0th3h3llam1** [comment]
What type of REST API are you looking to implement and what's the scope? I'm interested in creating API for this project.

**Comment by mrmn2** [comment]
Hi @wh0th3h3llam1, thanks for reaching out. I wanted to implement the REST API with `Django Ninja` as it is lightweight and fast. However, I am not fixated on it. If you are experienced with the `Django Rest Framework` then this is also an option. 

At some point (not necessarily at once) the API should have the following features:
* Authentication should be done via a token, that can be created, deleted and recreated in the user settings.
* list all PDFs per user (possibly with filters, e.g. tags)
* get a PDF
* create/update/delete PDFs
* share a PDF
* list all shared PDFs per user
* get a shared PDF
* create/update/delete shared PDFs
* get all highlights / comments
* get highlights / comments of a PDF 

Basically most things that are possible via the UI should also be doable via the API. I have probably forgotten some things in the list above. If you have other suggestions I am open for your ideas.
